Spatial_Engine/Query/PointLayout.cs

```diff
@@ -0,0 +1,419 @@
+                        int div = divisions[i] + 1; 
+
+                        List<Point> subPts = subCurves[i].SamplePoints(div);
+
+                        //Remove points at start/end
+                        subPts.RemoveAt(subPts.Count - 1);
+                        subPts.RemoveAt(0);
+                        pts.AddRange(subPts);
+                    }
+
+                }
+                pts = pts.ISortAlongCurve(hostRegionCurve);
+                return pts;
+            }
+            else
+            {
+                int divisions = layout2D.NumberOfPoints;
+                bool closed = hostRegionCurve.IIsClosed();
+
+                if (!closed)
+                    divisions--;
+
+                List<Point> pts = hostRegionCurve.SamplePoints(divisions);
+
+                //Remove duplicate endpoint
+                if (closed)
+                    pts.RemoveAt(pts.Count - 1);
+
+                return pts;
+            }
+        }
+
+        /***************************************************/
+
+        [Description("Returns Points along a straight line through the region curve. If the line is discontinuous because of openings or concave host region, Points will be distributed to the different segments based on their length.")]
+        [Input("layout2D", "The layout object to query the points from.")]
+        [Input("hostRegionCurve", "The region curve of the objects associated with the layout.")]
+        [Input("openingCurves", "Optional opening curves in the region.")]
+        [Output("points", "Point layout generated by the layout objects and region curve.")]
+        public static List<Point> PointLayout(this LinearLayout layout2D, ICurve hostRegionCurve, List<ICurve> openingCurves = null)
+        {
+            Point refPoint = ReferencePoint(hostRegionCurve, openingCurves, layout2D.ReferencePoint);
+            refPoint = refPoint + layout2D.Offset * AlignOffsetVector(Vector.ZAxis.CrossProduct(layout2D.Direction), refPoint, hostRegionCurve.IBounds().Centre());
+            Line axis = new Line { Start = refPoint, End = refPoint + layout2D.Direction, Infinite = true };
+
+            List<Line> distributionLines = IntersectionLines(hostRegionCurve, openingCurves, axis);
+
+            if (distributionLines.Count == 0)
+            {
+                Engine.Reflection.Compute.RecordError("Count not find extents of distribution for the layout.");
+                return new List<Point>();
+            }
+
+            return DivisionPoints(distributionLines, layout2D.NumberOfPoints);
+        }
+
+        /***************************************************/
+
+
+        [Description("Returns Points along straight lines throughout the region. If the line is segemented by openings or the region curve the points will be distributed to the different segments based on their length. \n" +
```

Comment:
```suggestion
        [Description("Returns Points along a straight line through the region curve. If the line is discontinuous because of openings or concave host region, Points will be distributed to the different segments based on their length.\n" +
```